src/commands/account/view_account_summary/mod.rs

```diff
@@ -124,32 +124,59 @@ fn get_account_inquiry(
+
     let delegated_stake: color_eyre::Result<
         std::collections::BTreeMap<near_primitives::types::AccountId, near_token::NearToken>,
     > = match validators {
-        Ok(validators) => Ok(runtime.block_on(
-            futures::stream::iter(validators)
-                .map(|validator_account_id| async {
-                    let balance = get_delegated_staked_balance(
-                        &json_rpc_client,
-                        block_reference,
-                        &validator_account_id,
-                        account_id,
-                    )
-                    .await?;
-                    Ok::<_, color_eyre::eyre::Report>((validator_account_id, balance))
-                })
-                .buffer_unordered(concurrency)
-                .filter(|balance_result| {
-                    futures::future::ready(if let Ok((_, balance)) = balance_result {
-                        !balance.is_zero()
-                    } else {
-                        true
-                    })
-                })
-                .try_collect(),
-        )?),
+        Ok(validators) => {
+            let mut all_results = std::collections::BTreeMap::new();
+            let validators: Vec<_> = validators.into_iter().collect();
+
+            for (batch_index, validator_batch) in validators
+                .chunks(batch_size)
+                .map(|x| x.to_vec())
+                .enumerate()
+            {
+                if batch_index > 0 {
+                    // Wait 30 seconds before starting next batch
+                    tracing::info!(
+                        "Waiting for 30 seconds before fetching next batch of stake information"
+                    );
+                    runtime.block_on(async { tokio::time::sleep(batch_cooldown).await });
+                }
+
+                let batch_results = runtime.block_on(
+                    futures::stream::iter(validator_batch)
+                        .map(|validator_account_id| async {
+                            let balance = get_delegated_staked_balance(
+                                &json_rpc_client,
+                                block_reference,
+                                &validator_account_id,
+                                account_id,
+                            )
+                            .await?;
+                            Ok::<_, color_eyre::eyre::Report>((validator_account_id, balance))
+                        })
+                        .buffer_unordered(concurrency)
+                        .filter(|balance_result| {
+                            futures::future::ready(if let Ok((_, balance)) = balance_result {
+                                !balance.is_zero()
+                            } else {
+                                true
+                            })
+                        })
+                        .try_collect::<std::collections::BTreeMap<_, _>>(),
+                )?;
```

Comment:
Right now if one batch fails, then everything else fails as well. I'm not sure if this is okay, let's at least add a retry logic. Ideally, it'd be great to notify user if one batch failed after retries and move on to the next batch. What do you think about it?

Reply:
The intial version is hard to understand without running, but looks like it fails too on any error within a single initial batch.
With batching this simple command will take at least 30 seconds now on mainnet, and retries could make this even longer. Suggest skipping displaying delegated_stake field instead of retrying if there has been at least one error, but displaying other info, that was successfully queried